format/riff/dolby.go

```diff
@@ -0,0 +1,294 @@
+package riff
+
+// Dolby Metadata, e.g. Atmos, AC3, Dolby Digital [Plus]
+// https://tech.ebu.ch/files/live/sites/tech/files/shared/tech/tech3285s6.pdf
+// https://github.com/DolbyLaboratories/dbmd-atmos-parser
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func dbmdDecode(d *decode.D, size int64) any {
+	version := d.U32()
+	major := (version >> 24) & 0xFF
+	minor := (version >> 16) & 0xFF
+	patch := (version >> 8) & 0xFF
+	build := version & 0xFF
+	d.FieldValueStr("version", fmt.Sprintf("%d.%d.%d.%d", major, minor, patch, build))
+
+	d.FieldArray("metadata_segments", func(d *decode.D) {
+		for {
+			d.FieldStruct("metadata_segment", func(d *decode.D) {
+				segmentID := d.FieldU8("metadata_segment_id")
+
+				// TODO(jmarnell): I think I need a loop until, but not creating these empty segments
+				// spec says we're done with 0 ID, so I'd like to not make the empty segment(s)
+				if segmentID == 0 {
+					if d.BitsLeft() > 0 {
+						d.SeekRel(d.BitsLeft() * 8)
+					}
+					return
+				}
+
+				segmentSize := d.FieldU16("metadata_segment_size")
```

Comment:
Just a note: spec says "Valid range: 0 - 65535 (0 is interpreted as 65535 byte)." but it seems mediainfo does not care about  0 but do truncate the size